src/TableResizePlugin.js

```diff
@@ -165,41 +167,61 @@ function handleMouseLeave(view: EditorView): void {
 
 // Function that handles the mousedown event from the table cell.
 function handleMouseDown(view: EditorView, event: MouseEvent): boolean {
+  // It's possible that the resize action that happened earlier was inturrupted
+  // while its dependent mouse events were stopped or prevented by others.
+  // We need to stop the previous resize action if it did not finish.
+  cancelDrag && cancelDrag(event);
+
   const resizeState = PLUGIN_KEY.getState(view.state);
   if (resizeState.cellPos === -1 || resizeState.draggingInfo) {
     return false;
   }
 
-  view.dispatch(
-    view.state.tr.setMeta(PLUGIN_KEY, {
-      setDraggingInfo: calculateDraggingInfo(view, event, resizeState),
-    })
-  );
-
-  // Move events should be batched to avoid over-handling the mouse
-  // event.
-  const dragMove = batchMouseHandler(handleDragMove);
+  let dragStarted = false;
+  let dragMoveHandler = null;
 
   const finish = (event: MouseEvent) => {
     window.removeEventListener('mouseup', finish, true);
     window.removeEventListener('mousemove', move, true);
-    handleDragEnd(view, event);
+    dragStarted && handleDragEnd(view, event);
+    cancelDrag = null;
   };
 
   const move = (event: MouseEvent) => {
     if (event.which) {
-      dragMove(view, event);
+      if (!dragStarted) {
+        handleDragStart(view, event);
+        dragStarted = true;
+      }
+      // Move events should be batched to avoid over-handling the mouse
+      // event.
+      dragMoveHandler = dragMoveHandler || batchMouseHandler(handleDragMove);
+      dragMoveHandler(view, event);
     } else {
       finish(event);
     }
   };
 
+  cancelDrag = finish;
   window.addEventListener('mouseup', finish, true);
   window.addEventListener('mousemove', move, true);
   event.preventDefault();
   return true;
 }
 
+function handleDragStart(view: EditorView, event: MouseEvent): void {
+  const resizeState = PLUGIN_KEY.getState(view.state);
```

Comment:
I don't really understand the what the resize state variable is, so taking a blind leap of faith and assuming this works out

Reply:
- `PLUGIN_KEY.getState()` returns the "resize state" of the current editor state.  
- API DOC: https://prosemirror.net/docs/ref/version/0.20.0.html#state.Plugin.getState

Here the resize state contains two members:
// a bunch of static & read-only information to assist the interaction. 
-  draggingInfo:  DraggingInfo, 
// whether user is currently resizing the left-most border.
-  forMarginLeft: boolean,

In ProseMirror, the table is treated as a primitive value which might be shared by multiple instances of editor. To really the state of the plugin for the current editor's instance, we'd use `PLUGIN_KEY.getState()` .